From: Christoph Lameter <clameter@sgi.com>
Date: Fri, 14 Apr 2006 01:23:53 +0000 (-0700)
Subject: Setup an IA64 specific reclaim distance
X-git-tag: v2.6.17-rc3
X-git-url: http://www.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=e5ecc192dfc5e0b325dd8c99ce4c755714c9acbf
Patch-mainline: 2.6.17-rc3
References: 176145

[IA64] Setup an IA64 specific reclaim distance

RECLAIM_DISTANCE is checked on bootup against the SLIT table distances.
Zone reclaim is important for system that have higher latencies but not for
systems that have multiple nodes on one motherboard and therefore low latencies.

We found that on motherboard latencies are typically 1 to 1.4 of local memory
access speed whereas multinode systems which benefit from zone reclaim have
usually more than 1.5 times the latency of a local access.

Set the reclaim distance for IA64 to 1.5 times.

Signed-off-by: Christoph Lameter <clameter@sgi.com>
Signed-off-by: Tony Luck <tony.luck@intel.com>
Acked-by: Raymund Will <rw@suse.de>

---

Index: linux/include/asm-ia64/topology.h
===================================================================
--- linux.orig/include/asm-ia64/topology.h	2006-03-19 23:53:29.000000000 -0600
+++ linux/include/asm-ia64/topology.h	2006-05-16 19:11:46.054539038 -0500
@@ -23,6 +23,11 @@
 #define PENALTY_FOR_NODE_WITH_CPUS 255
 
 /*
+ * Distance above which we begin to use zone reclaim
+ */
+#define RECLAIM_DISTANCE 15
+
+/*
  * Returns the number of the node containing CPU 'cpu'
  */
 #define cpu_to_node(cpu) (int)(cpu_to_node_map[cpu])

